### PR TITLE
Manage session

### PR DESCRIPTION
**Pull request by afmicc**
#### Description

* Create private routes
* Redirect to `welcome` when the user is logged in
* Persist only user info at the local storage

#### Update

* Persist only `user > auth` data
* User absolute imports

### CODE REVIEW

File: src/components/privateRoute/PrivateRoute.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { Route, Redirect } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import routes from '../../constants/routes';
+
+const PrivateRoute = ({ component: Component, ...props }) => {
+  const authenticated = useSelector(state => state.user.auth);
+
+  console.log('authenticated', authenticated);
```
Comment (manema) [inline src/components/privateRoute/PrivateRoute.js]: Please delete this.

File: src/components/privateRoute/PrivateRoute.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import routes from '../../constants/routes';
+
+const PrivateRoute = ({ component: Component, ...props }) => {
+  const authenticated = useSelector(state => state.user.auth);
```
Comment (manema) [inline src/components/privateRoute/PrivateRoute.js]:
```suggestion
  const authenticated = useSelector(({ user: { auth }}) => auth);
```

Just a simple aesthetic enhancement

File: src/components/routes.js [hunk 3]
```diff
@@ -8,6 +8,6 @@ export default [
   { path: '/', component: HomePage, exact: true },
   { path: '/about', component: AboutPage },
```
Comment (manema) [inline src/components/routes.js]: What's about to use the routes you have defined as a constants in src/constants/routes.js ?

**Review by manema — APPROVED** [review]
looking good!! I have never used redux-persist so that I didn't take a look on that.

File: src/redux/configureStore.js [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+  const persistConfig = { key: 'targetMVD', storage, whitelist: ['user'] };
 
-  return createStore(
-    rootReducer,
+  let store = createStore(
+    persistReducer(persistConfig, rootReducer),
     initialState,
     composeEnhancers(applyMiddleware(thunk, reduxImmutableStateInvariant()))
   );
+  let persistor = persistStore(store);
```
Comment (peter7z) [inline src/redux/configureStore.js]: Both store and persistor should be declared as `const`

File: src/components/privateRoute/PrivateRoute.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  console.log('authenticated', authenticated);
+  console.log('props', props);
```
Comment (peter7z) [inline src/components/privateRoute/PrivateRoute.js]: Remove console logs!

**Review by peter7z — APPROVED** [review]
👏

**Pull request merged**